src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -1239,15 +1242,17 @@ boolean next() {
       Digest commandBlobDigest, Executor executor, RequestMetadata requestMetadata) {
     Supplier<ListenableFuture<Command>> fetcher =
         () -> notFoundNull(expect(commandBlobDigest, Command.parser(), executor, requestMetadata));
+
+    BiFunction<Digest, Executor, CompletableFuture<Command>> getCallback =
+        new BiFunction<Digest, Executor, CompletableFuture<Command>>() {
+          @Override
+          public CompletableFuture<Command> apply(Digest digest, Executor executor) {
+            return toCompletableFuture(fetcher.get());
+          }
```

Comment:
Should we not be using the supplied executor in the fetcher's call to expect at this point?

Reply:
moved fetcher inside of callback to use executor